tools/generator/src/Extensions/XCScheme+Extensions.swift

```diff
@@ -0,0 +1,227 @@
+            profileAction = .init(profileActionInfo: profileActionInfo)
+        } else {
+            profileAction = .init(
+                buildableProductRunnable: nil,
+                buildConfiguration: XCSchemeConstants.defaultBuildConfigurationName
+            )
+        }
+
+        self.init(
+            name: schemeInfo.name,
+            lastUpgradeVersion: XCSchemeConstants.defaultLastUpgradeVersion,
+            version: XCSchemeConstants.lldbInitVersion,
+            buildAction: buildAction,
+            testAction: testAction,
+            launchAction: launchAction,
+            profileAction: profileAction,
+            analyzeAction: .init(analyzeActionInfo: schemeInfo.analyzeActionInfo),
+            archiveAction: .init(archiveActionInfo: schemeInfo.archiveActionInfo),
+            wasCreatedForAppExtension: schemeInfo.wasCreatedForAppExtension ? true : nil
+        )
+    }
+}
+
+// MARK: XCScheme.BuildAction
+
+extension XCScheme.BuildAction {
+    convenience init(
+        buildMode: BuildMode,
+        buildActionInfo: XCSchemeInfo.BuildActionInfo
+    ) throws {
+        self.init(
+            buildActionEntries: buildActionInfo.targetInfos.buildActionEntries,
+            preActions: buildMode.usesBazelModeBuildScripts ?
+                try buildActionInfo.targetInfos.bazelBuildPreActions : [],
+            parallelizeBuild: true,
+            buildImplicitDependencies: true
+        )
+    }
+}
+
+// MARK: XCScheme.BuildAction.Entry
+
+extension XCScheme.BuildAction.Entry {
+    convenience init(withDefaults buildableReference: XCScheme.BuildableReference) {
+        self.init(
+            buildableReference: buildableReference,
+            buildFor: [
+                .running,
+                .testing,
+                .profiling,
+                .archiving,
+                .analyzing,
+            ]
+        )
+    }
+}
+
+// MARK: XCScheme.ExecutionAction
+
+extension XCScheme.ExecutionAction {
+    /// Initialize the output file for build with Bazel
+    static let initBazelBuildOutputGroupsFile = XCScheme.ExecutionAction(
+        scriptText: #"""
+mkdir -p "${BAZEL_BUILD_OUTPUT_GROUPS_FILE%/*}"
+if [[ -s "$BAZEL_BUILD_OUTPUT_GROUPS_FILE" ]]; then
+    rm "$BAZEL_BUILD_OUTPUT_GROUPS_FILE"
+fi
+"""#,
+        title: "Initialize Bazel Build Output Groups File"
+    )
+
+    /// Create an ExecutionAction that builds with Bazel
+    convenience init(
+        bazelBuildFor buildableReference: XCScheme.BuildableReference,
+        name: String,
+        hostIndex: Int?
+    ) {
+        let hostTargetOutputGroup: String
+        if let hostIndex = hostIndex {
+            hostTargetOutputGroup = #"""
+echo "b $BAZEL_HOST_TARGET_ID_\#(hostIndex)" >> "$BAZEL_BUILD_OUTPUT_GROUPS_FILE"
+"""#
+        } else {
+            hostTargetOutputGroup = ""
+        }
+
+        let scriptText = #"""
+echo "b $BAZEL_TARGET_ID" >> "$BAZEL_BUILD_OUTPUT_GROUPS_FILE"
+\#(hostTargetOutputGroup)
+"""#
```

Comment:
This needs a trailing newline.
```suggestion
        let scriptText = #"""
echo "b $BAZEL_TARGET_ID" >> "$BAZEL_BUILD_OUTPUT_GROUPS_FILE"
\#(hostTargetOutputGroup)\#

"""#
```